http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/util/GKEInterpreter.scala

```diff
@@ -541,7 +541,7 @@ class GKEInterpreter[F[_]: Parallel: ContextShift: Timer](
         } yield operationOpt
       }
 
-      _ <- operationOpt.traverse(_ => nodepoolQuery.markAsDeleted(params.nodepoolId, ctx.now).transaction)
+      _ <- nodepoolQuery.markAsDeleted(params.nodepoolId, ctx.now).transaction
```

Comment:
So, if I remember correctly, the issue stems from the NonLeo pubsub subscriber calling this function, and the nodepool not terminating in a deleted status. 

It seem like, if the nodepool is not actually deleted, we'd want to mark it as error and record the error in some way, not just mark it as deleted anyways. Unfortunately, we currently have an app-centric error recording model... 

The LeoPubsubMessageSubscriber has infrastructure where you can raise an error at any stage of message processing, and the source is at least recorded (i.e., `nodepool` being the source here) and logged in the app error table. The solution I'd imagine is 
1) we want to introduce something like this error bubbling up and being logged in `NonLeoMessageSubscriber`
2) we'd want to `.raiseError` in a `fold` on the operationOpt, essentially raising an error if the delete failed and logging it, so we can investigate later. Right now, and with the solution above, the error is snarfed and never seen again.